src/range.rs

```diff
@@ -39,47 +34,84 @@ impl fmt::Debug for TextRange {
     }
 }
 
-/// Creates a new `TextRange` with given `start` and `end.
+/// Creates a new `TextRange` with the given `start` and `end` (`start..end`).
 ///
 /// # Panics
 ///
 /// Panics if `end < start`.
 #[allow(non_snake_case)]
+#[inline]
 pub fn TextRange(start: TextSize, end: TextSize) -> TextRange {
     assert!(start <= end);
     TextRange { start, end }
 }
 
-/// Identity methods.
 impl TextRange {
-    /// Creates a zero-length range at the specified offset.
-    pub const fn empty(self, offset: TextSize) -> TextRange {
+    /// Create a zero-length range at the specified offset (`offset..offset`).
+    #[inline]
+    pub const fn empty(offset: TextSize) -> TextRange {
         TextRange {
             start: offset,
             end: offset,
         }
     }
 
+    /// Create a range up to the given end (`..end`).
+    #[inline]
+    pub const fn before(end: TextSize) -> TextRange {
+        TextRange {
+            start: TextSize::zero(),
+            end,
+        }
+    }
+
+    /// Create a range after the given start (`start..`).
+    ///
+    /// This returns a std [`RangeFrom`] rather than `TextRange` because
+    /// `TextRange` does not support right-unbounded ranges. As such, this
+    /// should only be used for direct indexing, and bounded ranges should be
+    /// used for persistent ranges (`TextRange(start, TextSize::of(text))`).
+    #[inline]
+    pub const fn after(start: TextSize) -> RangeFrom<usize> {
```

Comment:
Not sure about before/after. Like, they have different return type but similar name, which is very unboring.

Reply:
I removed `after`; we still probably need _some_ sort of answer for the functionality it exposes, though. `before` feels like probably the best possible name for its functionality, however.

Reply:
Should we just bite the bullet and add a `TextRangeFrom` type for this? Or is that too much for what should fundamentally be a niche operation?

How about even having `TextRangeFrom` and `TextRangeTo`, so `TextRange::before(size)` remains "everything before `size`" even after a coordinate space change?

Reply:
>Or is that too much for what should fundamentally be a niche operation?

It's definitely way to much imo.